airbyte-integrations/connectors/source-klaviyo/source_klaviyo/streams.py

```diff
@@ -224,18 +182,32 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
         else:
             return super().get_updated_state(current_stream_state, latest_record)
 
-    def read_records(
+    def stream_slices(
         self,
         sync_mode: SyncMode,
         cursor_field: Optional[List[str]] = None,
-        stream_slice: Optional[Mapping[str, Any]] = None,
         stream_state: Optional[Mapping[str, Any]] = None,
-    ) -> Iterable[StreamData]:
-        yield from super().read_records(sync_mode, cursor_field, stream_slice, stream_state)
-        yield from self.archived_stream.read_records(sync_mode, cursor_field, stream_slice, stream_state)
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+        return [{"archived": flag} for flag in (False, True)]
+
+    def request_params(
+        self,
+        stream_state: Optional[Mapping[str, Any]],
+        stream_slice: Optional[Mapping[str, Any]] = None,
+        next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> MutableMapping[str, Any]:
+        state = (stream_state or {}).get("archived") if stream_slice.get("archived") else stream_state
+        params = super().request_params(stream_state=state, stream_slice=stream_slice, next_page_token=next_page_token)
+        if stream_slice.get("archived"):
+            archived_filter = "equals(archived,true)"
+            if "filter" in params and archived_filter not in params["filter"]:
+                params["filter"] = f"and({params['filter']},{archived_filter})"
```

Comment:
I suspect this can also be done with interpolation. does this need to stay in Python?